### DISCUSSION TITLE
Sgd with decaying rate in Netket

### DISCUSSION THREAD

**Discussion by wttai004** [netket/netket#2002] [Q&A] [answered]
The documentation on netket.optimizer.Sgd suggests that there's a version of Sgd with decaying learning rate implemented in the code (see [documentation](https://netket.readthedocs.io/en/latest/api/_generated/optim/netket.optimizer.Sgd.html#netket.optimizer.Sgd)):

`
NetKet also implements two extensions to the simple SGD, the first one is L2 regularization, and the second one is the possibility to set a decay factor gamma for the learning rate, such that at iteration  the learning rate is gamma delta^n
`

However, I don't see anything in the code that matches this description. Is this documentation accurate? In general, what would be a good recommended optimizer algorithm? To my knowledge, Adam is most commonly used in CS machine learning but isn't compatiblel with stochastic reconfiguration.

**Comment by gcarleo** [comment, marked as answer]
hi, I guess that is a residual of old documentation that has not been removed. The current paradigm is just to use any optimizer from [optax ](https://optax.readthedocs.io/en/latest/) or some of the basis optimizers that are built in netket. For example if you want to have a linear schedule in the learning rate you can do something like 

``` python
op = nk.optimizer.Sgd(learning_rate=optax.linear_schedule(0.1, 0.0001, 500))
```

In general you can just use Adam from optax with L2 regularization etc. Just refer to their docs. 

> To my knowledge, Adam is most commonly used in CS machine learning but isn't compatiblel with stochastic reconfiguration.

SR is to be used with a simple SGD update, Adam+SR is a strange object that is now very well behaved. In any case, besides some very special circumstances, the best optimizer is SGD+SR, also because it is theoretically and practically superior to Adam.
